ethcore/call-contract/src/call_contract.rs

```diff
@@ -18,12 +18,18 @@
 
 use bytes::Bytes;
 use ethereum_types::Address;
+use types::header::Header;
 use types::ids::BlockId;
 
 /// Provides `call_contract` method
 pub trait CallContract {
 	/// Like `call`, but with various defaults. Designed to be used for calling contracts.
 	fn call_contract(&self, id: BlockId, address: Address, data: Bytes) -> Result<Bytes, String>;
+
+	/// Makes a constant call to a contract, at the beginning of the block corresponding to the given header, i.e. with
```

Comment:
Maybe dumb q: what is "constant" about the call? Isn't it called for every block of the override is in effect?

Reply:
It's "constant" in the sense that it doesn't create a transaction and modify the blockchain's actual state. Not sure what the right word for this is.

Reply:
Maybe "stateless" or "immutable"? But technically the contract ABI can allow a contract call that modifies the state. I think assuming that the call modifies the state would be more complete.

Reply:
Not sure what the best wording is. In [this blog post](https://blog.b9lab.com/calls-vs-transactions-in-ethereum-smart-contracts-62d6b17d0bc2?gi=3fb53ba75e85) it's called "local" and "read-only call", as opposed to "verified" and "potentially state-changing transaction".

(To clarify: This _can_ be used with calls that _would_ modify the state. It's just that it throws away the modified state instead of creating a transaction which, when mined, would actually change the blockchain's EVM state.)